docs/DeveloperGuide.adoc

```diff
@@ -616,12 +635,93 @@ _Use case ends_
 +
 _Use case ends_
 
+[discrete]
+=== Use case: Tagging a transaction
+
+*MSS*
+
+1. User requests to tag a specific transaction.
+2. THRIFT processes the request and adds the tag to the specific transaction.
++
+_Use case ends_
+
+*Extensions*
+
+[none]
+* 1a. The parameters specified by the user are not valid.
++
+[none]
+** 1a1. THRIFT displays an invalid parameter error to the user and the transaction is not updated.
++
+_Use case ends_
+
+
+[discrete]
+=== Use case: Untagging a transaction
+
+*MSS*
+
+1. User requests to untag a specific transaction.
+2. THRIFT processes the request and removes the tag from the specific transaction.
++
```

Comment:
Maybe you could consider specifying what happens on the user's end after untagging. For example, a "successfully removed" message.